### DISCUSSION TITLE
MinIO for milvus cluster? Should I setup an independent from cluster?

### DISCUSSION THREAD

**Discussion by martinenkoEduard** [milvus-io/milvus#18865] [Q&A and General discussion]
I want to try to deploy minIO on a cluster.
What is the best way to do this?

Should I setup an independent MinIO server? 
If milvus deployed on kubernetis, how does the data  is split  between nodes?

What if i have one node with a large HHD, and multiple mini nodes with little free hard disk?

**Comment by yhmo** [comment]
You can refer to minio documentation to know how to deploy a minion cluster.

When you create a collection in milvus, there is a parameter named "shard_num", this number is to define how to split data into pieces.
`collection = Collection(name=name, data=None, schema=schema, shards_num = 2)`

The shard_num by default is 2. When you call insert() to insert 1000 entities into milvus, the 1000 entities will be split into 2 pieces. And if you deployed 2 data nodes, then each data node will consume one piece. But if you have only one data node, the data node will consume all the pieces.

Data node store segments into minio, if your minio server has not enough disk space, the milvus server will report error or could not work.

**Reply by martinenkoEduard** [reply]
Can you deploy a separate (outside of Kubernetes server) minio? How do I do it?

**Reply by yhmo** [reply]
Not sure how to do it, I think the key point is you need to ensure the Milvus server can access minio address from the k8s cluster to the outside minio service. The minio address is configured in the milvus.yml:

```
minio:
  address: localhost # Address of MinIO/S3
  port: 9000   # Port of MinIO/S3
```
